packages/smooth_app/lib/helpers/product_compatibility_helper.dart

```diff
@@ -8,18 +8,57 @@ class ProductCompatibilityHelper {
 
   final MatchedProduct matchedProduct;
 
-  Color getBackgroundColor() {
-    switch (matchedProduct.status) {
-      case null:
-      case MatchedProductStatus.UNKNOWN:
-        return GREY_COLOR;
-      case MatchedProductStatus.NO:
-        return RED_COLOR;
-      case MatchedProductStatus.YES:
-        return LIGHT_GREEN_COLOR;
+  Color getHeaderBackgroundColor(bool darkMode) {
+    if (!darkMode) {
+      switch (matchedProduct.status) {
+        case null:
+        case MatchedProductStatus.UNKNOWN:
+          return LIGHT_GREY_COLOR;
+        case MatchedProductStatus.NO:
+          return RED_COLOR;
+        case MatchedProductStatus.YES:
+          return LIGHT_GREEN_COLOR;
+      }
+    } else {
+      switch (matchedProduct.status) {
+        case null:
+        case MatchedProductStatus.UNKNOWN:
+          return PRIMARY_GREY_COLOR;
+        case MatchedProductStatus.NO:
+          return RED_COLOR;
+        case MatchedProductStatus.YES:
+          return DARK_GREEN_COLOR;
+      }
```

Comment:
I suggest new methods `Color _getDarkColors(Status)` and  `Color _getLightColors(Status)`.

Reply:
Changes done.